src/classes/BDI_Donations.cls

```diff
@@ -441,14 +454,19 @@ public class BDI_Donations {
             if (dataImportField == donationRecordTypeName) {
                 continue;
             }
-            // only honor Stage if we aren't updating its payment
-            if (dataImportField == donationStage && dataImport.PaymentImported__c != null) {
-                continue;
-            }
-            // only honor Amount if we aren't updating its payment
-            if (dataImportField == donationAmount && dataImport.PaymentImported__c != null) {
-                continue;
+
+            // only look at Stage and Amount if we aren't applying a new Payment
+            if (dataImport.DonationImportStatus__c != System.Label.bdiMatchedApplyNewPayment) {
```

Comment:
I'm seeing a lot of conditionals around the System.Label.bdiMatchedApplyNewPayment distributed through the code ... maybe it's unavoidable, but might there be a way to group this logic more tightly?

Reply:
The labels are referenced a lot in this class, but used for assignment of values more often than within conditionals.  Let's discuss ways in which we could improve upon this!

Reply:
In context with the rest of the code I think it makes more sense this way - let's look at it but I don't think it's a blocker, just a thought

Reply:
@ptewson-sfdo Hey Patrick!  I refactored this if block, but didn't see any great ways to improve on the Label references in the rest of the class.  Let me know if you want to discuss further!